src/tasks/AotCompilerTask/MonoAOTCompiler.cs

```diff
@@ -191,12 +191,39 @@ public class MonoAOTCompiler : Microsoft.Build.Utilities.Task
     /// </summary>
     public string? CacheFilePath { get; set; }
 
+    /// <summary>
+    /// Passes additional, custom arguments to --aot
+    /// </summary>
+    public string? AotArguments { get; set; }
+
+    /// <summary>
+    /// Passes temp-path to the AOT compiler
+    /// </summary>
+    public string? TempPath { get; set; }
+
+    /// <summary>
+    /// Passes ld-name to the AOT compiler, for use with UseLLVM=true
+    /// </summary>
+    public string? LdName { get; set; }
+
+    /// <summary>
+    /// Passes ld-flags to the AOT compiler, for use with UseLLVM=true
+    /// </summary>
+    public string? LdFlags { get; set; }
```

Comment:
I wonder if it might be better to have a `ITaskItem? CommonAotArguments` with metadata which act as the key-value pairs to pass to `--aot`.

Reply:
`LdFlags` will contain a `;` on Android. So it seems like using item metadata would always have an issue?

Our usage is now:
```xml
<MonoAOTCompiler
    Triple="$(_Triple)"
    ToolPrefix="$(_ToolPrefix)"
    MsymPath="$(_MsymPath)"
    Assemblies="@(_MonoAOTAssemblies)"
    CompilerBinaryPath="$(_MonoAOTCompilerPath)"
    DisableParallelAot="$(_DisableParallelAot)"
    IntermediateOutputPath="$(IntermediateOutputPath)"
    LibraryFormat="So"
    Mode="$(AndroidAotMode)"
    OutputDir="$(IntermediateOutputPath)aot\"
    OutputType="Library"
    UseAotDataFile="false"
    UseLLVM="$(EnableLLVM)"
    LLVMPath="$(_LLVMPath)"
    LdName="$(_LdName)"
    LdFlags="$(_LdFlags)"
    WorkingDirectory="$(MSBuildProjectDirectory)"
    AotArguments="$(AndroidAotAdditionalArguments)">
  <Output TaskParameter="CompiledAssemblies" ItemName="_AotCompiledAssemblies" />
  <Output TaskParameter="FileWrites"         ItemName="FileWrites" />
</MonoAOTCompiler>
```

But maybe `$(AndroidAotAdditionalArguments)` could be passed in a different way? What would the XML look like for `ITaskItem? CommonAotArguments`?

Reply:
> `LdFlags will contain a ; on Android. So it seems like using item metadata would always have an issue?`

We won't split the metadata on `CommonAotArguments`.
As for the metadata names, we could either use the real argument name, like` <ld-flags>foo; bar</ld-flags>`, or we could have `<LdFlags>foo; bar</LdFlags>` and convert to the real name in the task.

```xml
  <CommonAotArguments Include="common">
    <ld-flags>foo;bar</ld-flags>
    <ld-name>xyz</ld-name>
  </CommonAotArguments>
```
or
```xml
  <CommonAotArguments Include="common">
    <LdFlags>foo;bar</LdFlags>
    <LdName>xyz</LdName>
  </CommonAotArguments>
```

And you could have a `<PassVerbatim>..</PassVerbatim>` metadata (or better name) for just any additional string to be passed to `--aot`.

Feel free to push back, I'm just thinking aloud.

Reply:
The weird part about using `@(CommonAotArguments)`, is that you put `Include=""` with any random word inside. You can't leave out `Include`, or that would be an empty item group. I also like the idea of having C# properties, as it checks for typos -- you'll get an MSBuild error if you put `LdlFlags` instead of `LdFlags`.

Adding something like `@(CommonAotArguments)` might be ok if there is some unknown set. At least for Android, these changes now include every parameter we use. Are there many more not represented here?